sched/signal/sig_dispatch.c

```diff
@@ -374,6 +339,76 @@ static FAR sigpendq_t *nxsig_add_pendingsignal(FAR struct tcb_s *stcb,
   return sigpend;
 }
 
+/****************************************************************************
+ * Name: nxsig_alloc_dyn_pending
+ *
+ * Description:
+ *   Dynamically allocate more pending signal and pending sigaction
+ *   structures, if there are no more left. Note that this leaves the
+ *   the critical section for the allocation. During that time it is
+ *   it is possible that structures are freed, or another signalling thread
+ *   allocates more structures. This is not an issue, any extra pending
+ *   structures are freed after they get used.
+ *
+ * Assumptions:
+ *   Called with g_sigpendingsignal and g_sigpendingaction locked by
+ *   critical section.
+ *
+ ****************************************************************************/
+
+static irqstate_t nxsig_alloc_dyn_pending(irqstate_t flags)
+{
+  if (!up_interrupt_context())
```

Comment:
need check idle_task too

Reply:
hm, why does idle task dispatch signals?

Reply:
idle task can't call kmm_malloc since kmm_malloc will try to hold the mutex.

Reply:
Yes, but this function cannot be called from idle thread; this was also not checked before. Maybe add DEBUGASSERT instead?

Reply:
I believe debugassert is proper here; *if* this was called from idle thread (which should never happen) and there were no available pending signals, it would just lead to a potential crash at the point a pending signal would need to be added...